solidity/dashboard/src/components/Modal.jsx

```diff
@@ -70,6 +91,11 @@ export const ModalContextProvider = ({ children }) => {
   const [isOpen, setIsOpen] = useState(false)
   const [modalOptions, setModalOptions] = useState(null)
   const awaitingPromiseRef = useRef()
+  const showModalWindowForMobileUsers = useCurrentWidthLowerThanThreshold(1000)
+  const [
+    modalWindowForMobileUsersDisplayed,
+    setModalWindowForMobileUsersDisplayed,
+  ] = useState(false)
```

Comment:
I think displaying the modal for mobile users should be handled in the `AppLayout` component. The `ModalContextProvider` is a tool for showing modal and it should not decide which modal will be displayed. It just provides an API to open/close a modal. I believe the `AppLayout` component is the place to handle this logic. Thougts?